### ISSUE TITLE
httpd use-after-free

### ISSUE THREAD

**Issue by avikivity** [scylladb/seastar#1376]
Running the httpd unit test in debug mode:

```
==152386==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60200005bfd8 at pc 0x000000534763 bp 0x7f438a988de0 sp 0x7f438a988dd8
READ of size 8 at 0x60200005bfd8 thread T2
Reactor stalled for 32 ms on shard 1. Backtrace: 0x53bf0 0xaea1c3 0xab96f3 0x939662 0x95377f 0x94eabc 0x94ed3a 0x954956 0x3cb4f 0x7f438ecfcbff 0x7f438ecf2b50 0x7f438ecf2c87 0x7f438ecf2c87 0x7f438ed01c14 0x7f438ed01fc7 0x7f438ecf2837 0x7f438ecf28d5 0x7f438ece0ec9 0x7f438ece329a 0xdbe20 0xdc369 0x34782 0xc3b07 0xc31a5 0xc428b 0x534762 0x53680a 0x4e25d9 0x4e2668 0x619ae8 0x6ce8b1 0x992bba 0x998408 0x99dcd2 0x9bd094 0xa4b787 0xa3b54b 0xa1e654 0x79fffa 0x8e7ed4 0x8b14c 0x10c9ff
kernel callstack:
Reactor stalled for 90 ms on shard 1. Backtrace: 0x53bf0 0xaea1c3 0xab96f3 0x939662 0x95377f 0x94eabc 0x94ed3a 0x954956 0x3cb4f 0x7f438ececcf8 0x7f438ecf1ef7 0x7f438ed0148e 0x7f438ed01e39 0x84692b 0x7f438ed02021 0x7f438ecf2837 0x7f438ecf28d5 0x7f438ece0ec9 0x7f438ece329a 0xdbe20 0xdc369 0x34782 0xc3b07 0xc31a5 0xc428b 0x534762 0x53680a 0x4e25d9 0x4e2668 0x619ae8 0x6ce8b1 0x992bba 0x998408 0x99dcd2 0x9bd094 0xa4b787 0xa3b54b 0xa1e654 0x79fffa 0x8e7ed4 0x8b14c 0x10c9ff
kernel callstack:
    #0 0x534762 in seastar::lw_shared_ptr<seastar::queue<seastar::connected_socket> >::operator=(seastar::lw_shared_ptr<seastar::queue<seastar::connected_socket> >&&) (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x534762)
    #1 0x53680a in seastar::lw_shared_ptr<seastar::queue<seastar::connected_socket> >::operator=(decltype(nullptr)) (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x53680a)
    #2 0x4e25d9 in seastar::loopback_connection_factory::destroy_shard(unsigned int) (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x4e25d9)
    #3 0x4e2668 in seastar::loopback_connection_factory::destroy_all_shards()::{lambda()#1}::operator()() const (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x4e2668)
    #4 0x619ae8 in seastar::future<void> seastar::futurize<void>::invoke<seastar::loopback_connection_factory::destroy_all_shards()::{lambda()#1}&>(seastar::loopback_connection_factory::destroy_all_shards()::{lambda()#1}&) (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x619ae8)
    #5 0x6ce8b1 in seastar::smp_message_queue::async_work_item<seastar::loopback_connection_factory::destroy_all_shards()::{lambda()#1}>::run_and_dispose() (/home/avi/seastar/build/debug/tests/unit/httpd_test+0x6ce8b1)
    #6 0x992bba in seastar::reactor::run_tasks(seastar::reactor::task_queue&) /home/avi/seastar/src/core/reactor.cc:2509
    #7 0x998408 in seastar::reactor::run_some_tasks() /home/avi/seastar/src/core/reactor.cc:2946
    #8 0x99dcd2 in seastar::reactor::do_run() /home/avi/seastar/src/core/reactor.cc:3115
    #9 0x9bd094 in operator() /home/avi/seastar/src/core/reactor.cc:4326
    #10 0xa4b787 in __invoke_impl<void, seastar::smp::configure(const seastar::smp_options&, const seastar::reactor_options&)::<lambda()>&> /usr/include/c++/12/bits/invoke.h:61
    #11 0xa3b54b in __invoke_r<void, seastar::smp::configure(const seastar::smp_options&, const seastar::reactor_options&)::<lambda()>&> /usr/include/c++/12/bits/invoke.h:111
    #12 0xa1e654 in _M_invoke /usr/include/c++/12/bits/std_function.h:290
    #13 0x79fffa in std::function<void ()>::operator()() const /usr/include/c++/12/bits/std_function.h:591
    #14 0x8e7ed4 in seastar::posix_thread::start_routine(void*) /home/avi/seastar/src/core/posix.cc:73
    #15 0x7f438d4ae14c in start_thread (/lib64/libc.so.6+0x8b14c)
    #16 0x7f438d52f9ff in clone3 (/lib64/libc.so.6+0x10c9ff)
```

**Comment by tchaikov** [comment]
should be fixed by #1374 1374

**Comment by avikivity** [comment]
Fixed by 35c5927.

**Comment by avikivity** [comment]
Many thanks.
